mx_fetchjdk.py

```diff
@@ -0,0 +1,165 @@
+
+    print('export JAVA_HOME={}'.format(abspath(full_jdk_path)))
+
+
+def _parse_fetchjdk_settings(args):
+    settings = {}
+    settings["quiet"] = False
+    settings["keep-archive"] = False
+    settings["jdk-path"] = abspath(join('bin', 'jdks'))
+
+    jdk_paths = find_system_jdks()
+    if len(jdk_paths) > 0:
+        settings["jdk-path"] = jdk_paths[0]
+        found_jdk_path = True
+    else:
+        found_jdk_path = False
+
+    common_location = join(_mx_home, 'common.json')
+
+    parser = ArgumentParser(prog='mx fetch-labsjdk')
+    parser.add_argument('--java-distribution', action='store', help='JDK distribution that should be downloaded (e.g., "labsjdk-ce-11" or "openjdk8")')
+    parser.add_argument('--configuration', action='store', help='location of configuration json file (default: \'{}\')'.format(common_location))
+    parser.add_argument('--to', action='store', help='location where JDK would be downloaded (default: \'{}\')'.format(settings["jdk-path"]))
+    parser.add_argument('--alias', action='store', help='name of symlink to JDK')
+    parser.add_argument('--keep-archive', action='store_true', help='keep downloaded JDK archive')
+    parser.add_argument('-q', '--quiet', action='store_true', help='suppress logging output')
+    parser.add_argument('remainder', nargs=REMAINDER, metavar='...')
+    args = parser.parse_args(args)
+
+    if args.quiet is not None:
+        settings["quiet"] = args.quiet
+
+    if args.to is not None:
+        settings["jdk-path"] = args.to
+    elif not found_jdk_path and not settings["quiet"]:
+        mx.warn("No standard JDK location. Using {}".format(settings["jdk-path"]))
+
+    try:
+        if not exists(settings["jdk-path"]):
+            os.makedirs(settings["jdk-path"])
+        test_location = join(settings["jdk-path"], "test")
+        test_file = open(test_location, 'w')
+        test_file.close()
+        os.remove(test_location)
+    except (IOError, OSError):
+        mx.abort("Path '"+settings["jdk-path"]+"' is not writable. " + os.linesep +
+        "Rerun command with elevated privileges, or choose different JDK download location.")
+
+    if args.configuration is not None:
+        common_location = args.configuration
+    else:
+        if mx.primary_suite() is not None:
+            common_location = join(mx.primary_suite().vc_dir, 'common.json') # Try fetching suite config
+        else:
+            common_location = join(os.getcwd(), 'common.json') # Fallback to same folder
+            if not exists(common_location):
+                common_location = join(_mx_home, 'common.json') # Fallback to mx
+            mx.warn("Selected `{}` as configuration location, since no location is provided".format(common_location))
+    if not exists(common_location):
+        mx.abort("Configuration file doesn't exist")
+
+    JdkDistribution.parse_common_json(common_location)
+
+    if args.java_distribution is not None:
```

Comment:
I think it there would be less redundancy if all of this was done by the options parser.